Sources/AppcuesKit/Presentation/Public/ExperiencePluginConfiguration.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  ExperiencePluginConfiguration.swift
+//  AppcuesKit
+//
+//  Created by James Ellis on 12/19/22.
+//  Copyright © 2022 Appcues. All rights reserved.
+//
+
+import Foundation
+
+@objc
+public class ExperiencePluginConfiguration: NSObject {
+
+    private var decoder: PluginDecoder
+
+    init(_ decoder: PluginDecoder) {
+        self.decoder = decoder
+    }
+
+    public func decode<T: Decodable>(_ type: T.Type) -> T? {
+        return decoder.decode(type)
+    }
+}
```

Comment:
the new init object, with the `PluginDecoder` injection that allows it to be used across Actions and Traits, as well as more easily tested with mock decoder.